libmamba/src/core/fetch.cpp

```diff
@@ -640,8 +697,33 @@ namespace mamba
 
     bool MultiDownloadTarget::download(bool failfast)
     {
+        auto& ctx = Context::instance();
+
+        if (m_targets.empty())
+        {
+            LOG_INFO << "All targets to download are cached";
+            return true;
+        }
+
+        std::sort(
```

Comment:
is this to optimize the download order?

Reply:
yes, it is a basic attempt to optimize download speed by starting large downloads first.
this could be part of another PR if you prefer, and should be probably improved by implementing several other strategies. I'm thinking about a more global optimization of installation process where some package (some packages are downloaded while others are extracting or (un-)linking)